sarathi/model_executor/attention/base_attention_wrapper.py

```diff
@@ -37,6 +38,32 @@ def get_timer(self, operation: OperationMetrics, layer_id: Optional[int] = None)
         if self._timers.get((operation, layer_id)) is None:
             self._timers[(operation, layer_id)] = CudaTimer(operation, layer_id)
         return self._timers.get((operation, layer_id))
+    
+    def init_gpu_cache(self, num_gpu_blocks: int) -> None:
+        gpu_cache: List[torch.Tensor] = []
+        self.num_gpu_blocks = num_gpu_blocks
+
+        for _ in range(self.num_layers):
+            gpu_blocks = self.get_cache_block(
+                self.num_gpu_blocks, dtype=self.dtype, device="cuda"
+            )
+            gpu_cache.append(gpu_blocks)
+        
+        self.gpu_cache = gpu_cache
+
+    def get_cache_block(self, num_blocks: int, **kwargs) -> torch.Tensor:
+        return torch.randn(
+            num_blocks,
+            2,
```

Comment:
i think get_cache_block needs to be abstract and get_cache_block_size can be defined in the base class? basically, the size of kv cache per token is a model property, so it would remain same irrespective of the backend, whereas, how the dimensions of the cache can depend on the backend?

Reply:
Understood. Moved get_cache_block into the child class and get_cache_block_size into base class.